File: LondonDataServices.IDecide.Manage.Server.Tests.Integration/Models/Consumers/Consumer.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace LondonDataServices.IDecide.Manage.Server.Tests.Integration.Models.Consumers
{
    public class Consumer
    {
        public Guid Id { get; set; }
        public string EntraId { get; set; }
        public string Name { get; set; }
        public string ContactPerson { get; set; }
        public string ContactNumber { get; set; }
        public string ContactEmail { get; set; }
        public string CreatedBy { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }
    }
}

File: LondonDataServices.IDecide.Portal.Client/src/models/consumers/consumer.ts
import { ConsumerAdoption } from '../consumerAdoptions/consumerAdoption';

export class Consumer {
    public id?: string;
    public name?: string;
    public accessToken?: string;
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;
    public consumerAdoptions?: ConsumerAdoption[];

    constructor(consumer?: Partial<Consumer>) {
        this.id = consumer?.id || '';
        this.name = consumer?.name || '';
        this.accessToken = consumer?.accessToken || '';
        this.createdBy = consumer?.createdBy || '';
        this.createdDate = consumer?.createdDate ? new Date(consumer.createdDate) : undefined;
        this.updatedBy = consumer?.updatedBy || '';
        this.updatedDate = consumer?.updatedDate ? new Date(consumer.updatedDate) : undefined;
        this.consumerAdoptions = consumer?.consumerAdoptions || [];
    }
}